components/style/servo/url.rs

```diff
@@ -126,3 +127,33 @@ impl ToCss for SpecifiedUrl {
         dest.write_str(")")
     }
 }
+
+impl ToComputedValue for SpecifiedUrl {
+    type ComputedValue = ComputedUrl;
+
+    // If we can't resolve the URL from the specified one, we fall back to the original
+    // but still return it as a ComputedUrl::Invalid
+    fn to_computed_value(&self, _: &Context) -> Self::ComputedValue {
+        match self.resolved {
+            Some(ref url) => ComputedUrl::Valid(url.clone()),
+            None => match self.original {
+                Some(ref url) => ComputedUrl::Invalid(url.clone()),
+                None => ComputedUrl::Invalid(Arc::new("about:invalid".to_string())),
```

Comment:
Right, but with your patch we're changing that, right? And the uncompute stuff (which is the `from_computed_value` implementation) now will always have one or the other.